Update ECK + Elastic Agent run as root >= 8.16 (#4089)
elastic/docs-content: deploy-manage/deploy/cloud-on-k8s/configuration-examples-standalone.md

## Storing local state in host path volume [k8s_storing_local_state_in_host_path_volume]

{{agent}} managed by ECK stores local state in a host path volume by default. This ensures that {{integrations}} run by the agent can continue their work without duplicating work that has already been done after the Pod has been recreated for example because of a Pod configuration change. Multiple replicas of an agent, for example {{fleet}} Servers, can not be deployed on the same underlying {{k8s}} node as they would try to use the same host path. There are 2 options for managing this feature that differ depending on the version of Elastic Agent.

::::{tab-set}

> :::{tab-item} {{agent}} 8.16 and higher

Reviewer: ```suggestion
:::{tab-item} {{agent}} 8.16 and later
```

> :::{applies-item} { "stack": "ga 8.16" } In Elastic Agent versions 8.16 and beyond:

Reviewer: looked again at this - this might be failing because applies-item doesn't accept freeform text - just version info

because this is stack 8.x, consider just using a regular tabset and just the titles - 

```
::::{tab-set}

:::{tab-item} {{agent}} 8.16 and higher
```	

we are waiting for some functionality in the applies tags to better indicate ranges that will let you use applies switch more readily here in future

1. If local state storage in `hostPath` volumes is not desired this can be turned off by configuring an `emptyDir` volume instead.

> 2. Run {{agent}} with a `hostPath` volume. No additional configuration is required.

Reviewer: These are not sequential steps, so you should not use a numbered list

```suggestion
* If local state storage in `hostPath` volumes is not desired this can be turned off by configuring an `emptyDir` volume instead.
* Run {{agent}} with a `hostPath` volume. No additional configuration is required.
```

:::

:::{tab-item} {{agent}} 8.15 and before 

1. If local state storage in `hostPath` volumes is not desired this can be turned off by configuring an `emptyDir` volume instead.
2. If local state storage is still desired but running the Agent container as root is not allowed, then you can run a `DaemonSet` that adjusts the permissions for the Agent local state on each Node prior to running {{agent}}. Note that this `DaemonSet` must be `runAsUser: 0` and possibly `privileged: true`. Also note the {{kib}} changes required to trust the {{es}} CA when running in fleet mode.

:::
::::

> Full configuration examples exist in  [Running as a non-root user](configuration-fleet.md#k8s-elastic-agent-running-as-a-non-root-user).

Reviewer: ```suggestion
For full configuration examples, refer to [Running as a non-root user](configuration-fleet.md#k8s-elastic-agent-running-as-a-non-root-user).
```
